### DISCUSSION TITLE
PCManFM-Qt - add "media info" properties options in Visible Columns setting

### DISCUSSION THREAD

**Discussion by CluelessTechnologist** [lxqt/lxqt#2320] [General]
What do you think about adding certain media info options to the visible columns setting in the Detalied List view? The scope of "media info" can be very large but what I personally miss is image dimensions and track info for music files.

Dolphin has this feature but ironically I wasn't able to get it working. Maybe I was affected by this [bug](https://bugs.kde.org/show_bug.cgi?id=429572).  But anyway here is some images of what I mean:

**PCManFM-Qt:**

![image](https://user-images.githubusercontent.com/37050524/199703643-bec37256-b0b7-4280-951a-ccb0afb7cf87.png)
![image](https://user-images.githubusercontent.com/37050524/199704287-3f8fc384-07f9-4911-b434-28a026f8ed17.png)

**Dolphin:** 

![image](https://user-images.githubusercontent.com/37050524/199703830-49f72b07-41c5-4575-994e-c03fdb45f556.png)
![image](https://user-images.githubusercontent.com/37050524/199704468-0faf7390-8d5c-48a7-a305-cde0d1347531.png)

Is this is something that would be in scope for the project or should it be implemented as a plugin?

**Comment by tsujan** [comment]
Practically speaking, LXQt follows Unix philosophy ("Make each program do one thing well..."). PCManFM-Qt isn't a media player and doesn't follow Dolphin's design.

However, PCManFM-Qt supports [Custom Actions](https://github.com/lxqt/pcmanfm-qt/wiki/custom_actions), which are very flexible and can be used for almost any purpose — showing media info included.

**Reply by CluelessTechnologist** [reply]
Yes, that's what I thought. It's supposed to be a minimal file manager after all. 
>However, PCManFM-Qt supports [Custom Actions](https://github.com/lxqt/pcmanfm-qt/wiki/custom_actions), which are very flexible and can be used for almost any purpose — showing media info included.

Forgive me if I'm wrong here but after reading about this functionality it seems like there's no way to get the output of the custom actions displayed in Visible Columns in PCManFM-QT under detailed view? Seems to me it's meant to be used for custom scripts on bulk items etc rather than displaying detailed information.

Maybe this is a better match for my use case? https://github.com/lxqt/pcmanfm-qt/issues/40

**Reply by tsujan** [reply]
>  It's supposed to be a minimal file manager after all.

Not minimal but uncluttered.

> t seems like there's no way to get the output of the custom actions displayed in Visible Columns in PCManFM-QT under detailed view?

Custom actions don't depend on any view mode:

![media_info](https://user-images.githubusercontent.com/981076/199993973-3a85327b-747f-428a-9871-67163d14ddee.png)

The goal is doing all kinds of extra jobs, and they do it most flexibly.

**Reply by CluelessTechnologist** [reply]
Well the problem is that will display the media info in a external application will it not? I'm looking to display image dimensions of all images when I browse my file system using PCManFM-Qt. Having to right click images to get image dimensions is not a quick enough overview for me. I have nothing against using a external application to get the media info but I need a way to display it in PCManFM-Qt.

**Reply by tsujan** [reply]
> Well the problem is that will display the media info in a external application will it not?

Exactly. Adding more dependencies to libfm-qt/pcmanfm-qt is not an option.

**Reply by CluelessTechnologist** [reply]
> Exactly. Adding more dependencies to libfm-qt/pcmanfm-qt is not an option.

I see, thanks for clarifying! I was really hoping for some sort of plugin support so there was a way to display the information in PCManFM-Qt. Because I really empathize with the Unix philosophy, I'm not a stranger to having to use external applications to customize things to fit my use case, but if there is no way to get the information into PCManFM-Qt I cannot use it in _this specific example_. But I am sure [Custom Actions](https://github.com/lxqt/pcmanfm-qt/wiki/custom_actions) works great for other use cases.
